scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -862,9 +890,25 @@ void BPF_STRUCT_OPS(layered_dispatch, s32 cpu, struct task_struct *prev)
 		}
 	}
 
-	if (scx_bpf_consume(HI_FALLBACK_DSQ))
+	/* consume any hi fallback dsqs starting with the llc local dsq */
+	dsq_id = cpu_hi_fallback_dsq_id(cpu);
+	if (scx_bpf_consume(dsq_id))
 		return;
 
+	/* consume from any hi fallback dsqs in the node */
+	bpf_for(idx, 0, nr_llcs) {
+		llc_id = llc_iter_cpu_offset(idx, cpu);
+		if (llc_node_id(llc_id) != node_id)
+			continue;
+
+		next_dsq_id = llc_hi_fallback_dsq_id(llc_id);
+		if (next_dsq_id == dsq_id)
+			continue;
+
+		if (scx_bpf_consume(next_dsq_id))
+			return;
+	}
```

Comment:
Can't we end up in a situation where a task which is affine to one node is queued on the fallback dsq of another node?

Reply:
Hmmm yeah... so is consuming from the llc local dsq sufficient then? Perhaps I need to rethink the load balancing a bit more.

Reply:
Hmm... isn't it in the other direction? We can end up in a situation where a task is queued on a fallback DSQ which will never be consumed by a CPU that can run that task.